imap_processing/codice/codice_l1a_lo_counters_aggregated.py

```diff
@@ -0,0 +1,258 @@
+"""CoDICE L1A Lo Counters aggregated processing functions."""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.codice import constants
+from imap_processing.codice.decompress import decompress
+from imap_processing.codice.utils import (
+    CODICEAPID,
+    ViewTabInfo,
+    calculate_acq_time_per_step,
+    get_codice_epoch_time,
+    get_counters_aggregated_pattern,
+    get_view_tab_info,
+    read_sci_lut,
+)
+from imap_processing.spice.time import met_to_ttj2000ns
+
+logger = logging.getLogger(__name__)
+
+
+def l1a_lo_counters_aggregated(
+    unpacked_dataset: xr.Dataset, lut_file: Path
+) -> xr.Dataset:
+    """
+    Process CoDICE Lo Counters aggregated L1A data.
+
+    Parameters
+    ----------
+    unpacked_dataset : xarray.Dataset
+        Unpacked dataset from L0 packet file.
+    lut_file : Path
+        Path to the LUT file for processing.
+
+    Returns
+    -------
+    xarray.Dataset
+        Processed L1A dataset for Hi Omni data.
+    """
+    # lookup in LUT table.
+    table_id = unpacked_dataset["table_id"].values[0]
+    view_id = unpacked_dataset["view_id"].values[0]
+    apid = unpacked_dataset["pkt_apid"].values[0]
+    plan_id = unpacked_dataset["plan_id"].values[0]
+    plan_step = unpacked_dataset["plan_step"].values[0]
+
+    logger.info(
+        f"Processing aggregated with - APID: {apid} / 0x{apid:X}, View ID: {view_id}, "
+        f"Table ID: {table_id}, Plan ID: {plan_id}, Plan Step: {plan_step}"
+    )
+    # ========== Get LUT Data ===========
+    # Read information from LUT
+    sci_lut_data = read_sci_lut(lut_file, table_id)
+
+    view_tab_info = get_view_tab_info(sci_lut_data, view_id, apid)
+    view_tab_obj = ViewTabInfo(
+        apid=apid,
+        view_id=view_id,
+        sensor=view_tab_info["sensor"],
+        three_d_collapsed=view_tab_info["3d_collapse"],
+        collapse_table=view_tab_info["collapse_table"],
+    )
+
+    if view_tab_obj.sensor != 0:
+        raise ValueError("Unsupported sensor ID for Lo processing.")
+
+    # ========== Get Voltage Data from LUT ===========
+    # Use plan id and plan step to get voltage data's table_number in ESA sweep table.
+    # Voltage data is (128,)
+    esa_table_number = sci_lut_data["plan_tab"][f"({plan_id}, {plan_step})"][
+        "lo_stepping"
+    ]
+    voltage_data = sci_lut_data["esa_sweep_tab"][f"{esa_table_number}"]
+
+    # ========= Decompress and Reshape Data ===========
+    if view_tab_obj.apid != CODICEAPID.COD_LO_INST_COUNTS_AGGREGATED:
+        raise ValueError("Unsupported APID for Lo Counters aggregated processing.")
```

Comment:
Didn't you pass in the apid to the view_tab_obj earlier, so can you ever encounter this case?